apps/ledger-live-mobile/src/components/RootNavigator/CustomImageNavigator.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useMemo } from "react";
+import { createStackNavigator } from "@react-navigation/stack";
+import { useTheme } from "styled-components/native";
+import { ScreenName } from "../../const";
+import { getStackNavigatorConfig } from "../../navigation/navigatorConfig";
+import Step1Cropping from "../../screens/CustomImage/Step1Crop";
+import Step2Preview from "../../screens/CustomImage/Step2Preview";
+import DebugScreen from "../../screens/CustomImage/DebugScreen";
+import Step3Transfer from "../../screens/CustomImage/Step3Transfer";
+import ErrorScreen from "../../screens/CustomImage/ErrorScreen";
+
+const Empty = () => null;
+
+export default function CustomImageNavigator() {
+  const { colors } = useTheme();
+  const stackNavigationConfig = useMemo(
+    () => getStackNavigatorConfig(colors, true),
+    [colors],
+  );
+
+  return (
+    <Stack.Navigator screenOptions={stackNavigationConfig}>
+      <Stack.Screen
+        name={ScreenName.CustomImageStep1Crop}
+        component={Step1Cropping}
+        options={{ title: "" }}
+      />
+      <Stack.Screen
+        name={ScreenName.CustomImageStep2Preview}
+        component={Step2Preview}
+        options={{ title: "" }}
+      />
+      <Stack.Screen
+        name={ScreenName.CustomImageStep3Transfer}
+        component={Step3Transfer}
+        options={{ title: "" }}
+      />
+      <Stack.Screen
+        name={ScreenName.CustomImageErrorScreen}
+        component={ErrorScreen}
+        options={{ title: "", headerLeft: Empty }}
+      />
+    </Stack.Navigator>
+  );
+}
+
+const Stack = createStackNavigator();
```

Comment:
Maybe with new navigators we should enforce typing ? (only CustomImageNavigator, not the BaseNavigator)

You can find an example on Ledger Live (not yet merged) [here](https://github.com/LedgerHQ/ledger-live/blob/aa76c034a6400add49b20d79aba9089681c235f7/apps/ledger-live-mobile/src/components/RootNavigator/SyncOnboardingNavigator.tsx#L21) - and the [doc](https://reactnavigation.org/docs/typescript/)

Reply:
Ah yes that's great 🔥 
I will do that

Reply:
Note: the `as` (ex: `name={ScreenName.DeviceModelSelection as "DeviceModelSelection"}`) is needed as long as `NavigatorName` is an Object and not an enum ... It has been changed by Thomas 1 month ago, but then reverted because one screen was not working (or something, don't really know why it was reverted)